testnet/stacks-node/src/mockamoto.rs

```diff
@@ -247,61 +558,177 @@ impl MockamotoNode {
-            self.mine_and_stage_block().unwrap();
-            info!("Processing a staging block");
-            self.process_staging_block().unwrap();
-            info!("Cycle done");
-            sleep(Duration::from_secs(5));
         }
+        Err(ChainstateError::NoSuchBlockError)
     }
 
-    fn produce_burnchain_block(&mut self) -> Result<(), ChainstateError> {
+    fn produce_burnchain_block(&mut self, initializing: bool) -> Result<(), BurnchainError> {
         let miner_pk = Secp256k1PublicKey::from_private(&self.miner_key);
         let miner_pk_hash = Hash160::from_node_public_key(&miner_pk);
 
         let parent_snapshot = SortitionDB::get_canonical_burn_chain_tip(&self.sortdb.conn())?;
-        let new_snapshot = make_snapshot(&parent_snapshot, &miner_pk_hash)?;
-        let mut sortdb_tx = self.sortdb.tx_handle_begin(&parent_snapshot.sortition_id)?;
-
-        sortdb_tx.append_chain_tip_snapshot(
-            &parent_snapshot,
-            &new_snapshot,
-            &vec![],
-            &vec![],
-            None,
-            None,
-            None,
-        )?;
+        info!("Mocking bitcoin block"; "parent_height" => parent_snapshot.block_height);
+        let burn_height = parent_snapshot.block_height + 1;
+
+        let mut ops = vec![];
+
+        if burn_height == 1 {
+            let mut txid = [2u8; 32];
+            txid[0..8].copy_from_slice((burn_height + 1).to_be_bytes().as_ref());
+            let key_register = LeaderKeyRegisterOp {
+                consensus_hash: ConsensusHash([0; 20]),
+                public_key: VRFPublicKey::from_private(&self.vrf_key),
+                memo: miner_pk_hash.as_bytes().to_vec(),
+                txid: Txid(txid),
+                vtxindex: 0,
+                block_height: burn_height,
+                burn_header_hash: BurnchainHeaderHash([0; 32]),
+            };
+            ops.push(BlockstackOperationType::LeaderKeyRegister(key_register));
+        } else if !initializing {
+            let mut txid = [1u8; 32];
+            txid[0..8].copy_from_slice((burn_height + 1).to_be_bytes().as_ref());
+            txid[8..16].copy_from_slice((0u64).to_be_bytes().as_ref());
+
+            let (parent_block_ptr, parent_vtxindex) =
+                if parent_snapshot.winning_block_txid.as_bytes() == &[0; 32] {
+                    (0, 0)
+                } else {
+                    (parent_snapshot.block_height.try_into().unwrap(), 0)
+                };
+
+            let parent_vrf_proof = NakamotoChainState::get_block_vrf_proof(
+                self.chainstate.db(),
+                &parent_snapshot.consensus_hash,
+            )
+            .map_err(|_e| BurnchainError::MissingParentBlock)?
+            .unwrap_or_else(|| VRFProof::empty());
+
+            let vrf_seed = VRFSeed::from_proof(&parent_vrf_proof);
+
+            let block_commit = LeaderBlockCommitOp {
+                block_header_hash: BlockHeaderHash([0; 32]),
```

Comment:
Pretty sure this field is getting validated now.  This should be the index block hash of the last tenure's tenure-start block.

Reply:
I updated this to be the parent block ID, but I would suggest that this is _not_ being validated currently: the coordinator absolutely accepted blocks with null block header hashes.

Reply:
> the coordinator absolutely accepted blocks with null block header hashes.

Oh interesting!  I'll see if I can fix this in my tenure-change validation PR.